### PR TITLE
WIP: Analytics bundle

### PR DESCRIPTION
**Pull request by thnkloud9**
Please let me know what you think of the approach.  I decided to use monolog so that we could use the leveling to possibly highlight potentially slow or resource intensive renders.  I wanted to save the view path in the anayltics_log table but I was not sure how to pass it to Logger/DatabaseHandler.php.  Any ideas there would be helpful.

### CODE REVIEW

**Comment by ahilles107** [comment]
Please check scrutinizer issues.

**Comment by thnkloud9** [comment]
I'm not sure how to solve the duplicate issues.  I've tried combining the event listeners into one class (either Processor/RequestProcessor or EventListener/RenderListener) and in both cases I can't seem to get the events to fire correctly to get all of the data needed to stop the stopwatch and log the record.  Any ideas?

File: src/SWP/AnalyticsBundle/Resources/config/services.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        tags:
+            - { name: monolog.processor, method: processRecord }
+            - { name: kernel.event_listener, event: kernel.controller, method: onKernelController}
+            - { name: kernel.event_listener, event: kernel.view, method: onKernelView }
+            - { name: kernel.event_listener, event: kernel.response, method: onKernelResponse }
+
+    monolog.processor.request:
+        class: SWP\AnalyticsBundle\Processor\RequestProcessor
+        calls:
+            - [ setRequest, ["@service_container"]]
```
Comment (takeit) [inline src/SWP/AnalyticsBundle/Resources/config/services.yml]: you can use `request_stack` service instead

**Comment by takeit** [comment]
I dont know if it worth to eliminate those duplicate issues on request listener. We will need to make an abstract listener to have not duplication issue a cross the whole project. Also I assume this bundle will be a separate repository, if this bundle will be included via composer you wont have those duplicates issues.

File: src/SWP/AnalyticsBundle/DataCollector/AnalyticsDataCollector.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+namespace SWP\AnalyticsBundle\DataCollector;
+
+use Symfony\Component\HttpKernel\DataCollector\DataCollector;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use SWP\AnalyticsBundle\Model\AnalyticsLog;
+
+class AnalyticsDataCollector extends DataCollector
+{
+    protected $_container;
```
Comment (takeit) [inline src/SWP/AnalyticsBundle/DataCollector/AnalyticsDataCollector.php]: variable name is not in camelcase

File: src/SWP/AnalyticsBundle/DataCollector/AnalyticsDataCollector.php [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+    protected $_em;
```
Comment (takeit) [inline src/SWP/AnalyticsBundle/DataCollector/AnalyticsDataCollector.php]: variable name is not in camelcase

File: src/SWP/AnalyticsBundle/EventListener/RenderListener.php [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\Stopwatch\Stopwatch; 
+
+use SWP\AnalyticsBundle\Controller\AnalyzedControllerInterface;
+
+class RenderListener extends WebProcessor
+{
+
+    protected $_stopwatch;
```
Comment (takeit) [inline src/SWP/AnalyticsBundle/EventListener/RenderListener.php]: variables names are not in camelcase (see also below)

File: src/SWP/AnalyticsBundle/Processor/RequestProcessor.php [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+ */
+namespace SWP\AnalyticsBundle\Processor;
+
+use Symfony\Bridge\Monolog\Processor\WebProcessor;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+class RequestProcessor extends WebProcessor
+{
+    protected $_request;
```
Comment (takeit) [inline src/SWP/AnalyticsBundle/Processor/RequestProcessor.php]: variable name not in camlcase

**Comment by ahilles107** [comment]
looks like there is a bug in code - check tests log.

**Comment by ahilles107** [comment]
implemented in #78

**Pull request closed without merging**